apps/frontend/src/app/2_molecules/AdvancedSettings/AdvancedSettings.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { FC, useMemo, useReducer } from 'react';
+
+import { t } from 'i18next';
+
+import { Accordion, AmountInput, ErrorBadge, ErrorLevel } from '@sovryn/ui';
+
+import { translations } from '../../../locales/i18n';
+
+type AdvancedSettingsProps = {
+  amount: string;
+  onChange: (value: string) => void;
+  invalid?: boolean;
+  className?: string;
+  errorMessage?: string;
+};
+
+export const AdvancedSettings: FC<AdvancedSettingsProps> = ({
+  amount,
+  onChange,
+  className,
+  invalid,
+  errorMessage,
+}) => {
+  const [open, toggle] = useReducer(v => !v, false);
+
+  const errorBadge = useMemo(() => {
+    if (!invalid || !errorMessage) {
```

Comment:
I think you need to provide some fallback `errorMessage` if it's not passed as a prop because you could end up in an error state without an error message and don't display anything. 

Another solution is to make `errorMessage` prop required (my preferred solution).